docs/ref/models/lookups.txt

```diff
@@ -94,6 +94,19 @@ following methods:
     ``compiler.compile(expression)`` should be used. The ``compiler.compile()``
     method will take care of calling vendor-specific methods of the expression.
 
+    ``kwargs`` may be used to supply extra context to the ``as_sql``-method
+    for that particular call. This can be useful for Expressions in
+    particular, where ``template`` can be overridden and variables intended
+    for interpolating the ``template``-string can be passed. 3rd-party
+    backends may have different SQL-syntax so they can pass these differences
+    through via ``kwargs``.
+
+    Example::
+
+        self.as_sql(compiler, connection,
+            template="%(function)s(%(arg)s)", function='Print', arg='Hello World'
+        )
```

Comment:
function -> method, correct? The actual code in django/db/models/functions.py is a bit different -- is the difference intentional? Maybe something like this could be nice:
```python
.. snippet::
    :filename: django/db/models/functions.py

    class ConcatPair(Func):
        ...
        def as_sqlite(self, compiler, connection):
            coalesced = self.coalesce()
            return super(ConcatPair, coalesced).as_sql(
                compiler, connection, template='%(expressions)s', arg_joiner=' || '
            )
```